prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirBundleUtils.kt

```diff
@@ -41,33 +41,68 @@ object FhirBundleUtils : Logging {
      * Converts a [value] of type [sourceType] into a compatible Base of type [targetType]. Returns the original value
      * and logs an error if the conversion is not supported.
      */
-    fun convertFhirType(value: Base, sourceType: String, targetType: String, logger: KotlinLogger = this.logger): Base {
+    fun convertFhirType(
+        value: Base,
+        sourceType: String,
+        targetType: String,
+        logger: KotlinLogger = this.logger,
+    ): Base? {
```

Comment:
No, I need it to be nullable so that returning null can be the distinguishing return value that indicates we need to try another type.

Reply:
Unless you have another idea of how to do this?

Reply:
Ahh that makes sense. No ideas really, maybe a throw but that's probably overkill.

Can we add a small note on the null return branch in the function comments?

Reply:
Oh wait, sorry, ti can be non-nullable. I was thinking about making ti recursive so then it would have needed to be nullable, but decided against it. Removing.